### PR TITLE
Make implementer tools aware of the config schema

### PR DESCRIPTION
**Pull request by brandones**
Also refactor the concept lookup box

conf

### CODE REVIEW

File: packages/esm-implementer-tools/src/configuration/value-editor.tsx [hunk 1]
```diff
@@ -1,18 +1,20 @@
...
 
 export default function ValueEditor({
-  valueString,
+  element,
   handleSave,
   handleClose,
 }: ValueEditorProps) {
   const ref = useRef(null);
-  const [tmpValue, setTmpValue] = useState(valueString);
+  const [tmpValue, setTmpValue] = useState(JSON.stringify(element._value));
```
Comment (FlorianRappl) [inline packages/esm-implementer-tools/src/configuration/value-editor.tsx:17]: We could use `() => JSON.stringify(element._value)` to only stringify when an initial state is demanded.

File: packages/esm-implementer-tools/src/configuration/concept-search.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        setConceptName(data.name.display);
+      });
+    }
+  }, [activeConceptUuid]);
+
+  useEffect(() => {
+    const ac = new AbortController();
+
+    if (searchTerm && searchTerm.length >= 2) {
+      performConceptSearch(searchTerm).then(({ data: { results } }) => {
```
Comment (FlorianRappl) [inline packages/esm-implementer-tools/src/configuration/concept-search.tsx:51]: Should `ac` be used here?

File: packages/esm-implementer-tools/src/configuration/concept-search.tsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    setSearchResults([]);
+  };
+
+  const handleSearchTermChange = debounce((searchTerm) => {
+    setSearchTerm(searchTerm);
+  }, searchTimeoutInMs);
+
+  useEffect(() => {
+    if (activeConceptUuid) {
+      const conceptString = activeConceptUuid;
```
Comment (FlorianRappl) [inline packages/esm-implementer-tools/src/configuration/concept-search.tsx:39]: Why the alias?

File: packages/esm-implementer-tools/src/configuration/concept-search.tsx [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+export function ConceptSearchBox({ setConcept }: ConceptSearchBoxProps) {
+  const [conceptName, setConceptName] = useState("");
+  const [searchTerm, setSearchTerm] = useState("");
+  const [searchResults, setSearchResults] = useState([]);
+  const [activeConceptUuid, setActiveConceptUuid] = useState<any>("");
+  const searchTimeoutInMs = 300;
+
+  const id = useMemo(() => uniqueId(), []);
+
+  const handleUuidChange = (concept) => {
```
Comment (FlorianRappl) [inline packages/esm-implementer-tools/src/configuration/concept-search.tsx:22]: Should we use `useCallback` here?

Same for the other handlers.

**Review by FlorianRappl — COMMENTED** [review]
LGTM :beers:

File: packages/esm-implementer-tools/src/configuration/concept-search.tsx (hunk 3 above)
Comment (brandones) [reply packages/esm-implementer-tools/src/configuration/concept-search.tsx:39]: This whole component was copy-pasted out of EditableValue (not to exonerate myself—I reviewed the original code!). I'll fix it now, anyway.

**Comment by jonathandick** [comment]
Brandon do you mind adding a few comments on the significance of this ticket? why it was important? what now becomes possible because of this.

File: packages/esm-implementer-tools/src/configuration/concept-search.tsx (hunk 4 above)
Comment (brandones) [reply packages/esm-implementer-tools/src/configuration/concept-search.tsx:22]: I don't think so—creating the function is not itself expensive, and re-creating the function shouldn't cause any unnecessary re-renders (I think—feel free to correct me here).

I have consolidated the other function into this one though.

**Comment by brandones** [comment]
@jonathandick With this work, we can show the default, description, and validators for each config element directly in the interface.

Once we add a `type` property to config elements, we can make implementer-tools do specific fancy things for different types of config elements. i.e. concept search for concepts (until then, for the purpose of this PR, I have implemented `isConcept` as a completely evil and fragile and temporary hack).

**Pull request merged**